### ISSUE TITLE
Passing dictionary argument to BatchRunner

### ISSUE THREAD

**Issue by devforfu** [projectmesa/mesa#307]
Not sure if it is an issue and not intended behaviour but nevertheless decided to notify about it.

When you pass a dictionary value into batch runner it only takes keys and not values. So you'll not get a cartesian product of your values. For example, if you pass something like this:

``` python
class MoneyModel(Model):

    def __init__(self, n, grid_params):  # grid_params is intended to be a dictionary
        super().__init__(self)
        ...


def main():
    ...
    n_iters, n_steps = 5, 200
    params = {
        "grid_params": {"height": 10, "width": 10},
        "n": range(10, 500, 10)}
    batch = BatchRunner(MoneyModel, params,
        iterations=n_iters, max_steps=n_steps,
        model_reporters={"Gini": compute_gini})
    batch.run_all()
```

You'll get issue in there: https://github.com/projectmesa/mesa/blob/master/mesa/batchrunner.py#L71, i.e. variable `kwargs` will be equal to `{'grid_params': 'width', 'n': 10}` (or any other dictionary key) instead of original value.

**Comment by jackiekazil** [comment]
Interesting. I didn't notice this before. This was not on purpose. I just think that no one noticed it before. 

Do you want to submit a PR for it? (If not, would you mind writing a test for it and someone else can code against?)

**Comment by devforfu** [comment]
Yes, sure, I'll do it.

**Comment by njvrzm** [comment]
This should be fixed by #374!
